base/commands/migration/start_stages_it_test.go

```diff
@@ -0,0 +1,133 @@
+//go:build migration
+
+package migration_test
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"testing"
+	"time"
+
+	_ "github.com/hazelcast/hazelcast-commandline-client/base"
+	_ "github.com/hazelcast/hazelcast-commandline-client/base/commands"
+	"github.com/hazelcast/hazelcast-commandline-client/base/commands/migration"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+func TestMigration(t *testing.T) {
+	testCases := []struct {
+		name string
+		f    func(t *testing.T)
+	}{
+		{name: "start_Successful", f: startTest_Successful},
+		{name: "start_Failure", f: startTest_Failure},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, tc.f)
+	}
+}
+
+func startTest_Successful(t *testing.T) {
+	tcx := it.TestContext{T: t}
+	ctx := context.Background()
+	tcx.Tester(func(tcx it.TestContext) {
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go tcx.WithReset(func() {
+			defer wg.Done()
+			Must(tcx.CLC().Execute(ctx, "start", "dmt-config", "--yes"))
+		})
+		successfulRunner(tcx, ctx)
+		tcx.AssertStdoutContains(`
+Hazelcast Data Migration Tool v5.3.0
+(c) 2023 Hazelcast, Inc.
+	
+Selected data structures in the source cluster will be migrated to the target cluster.	
+
+
+ OK   [1/3] Connected to the migration cluster.
+first message
+ OK   [2/3] Started the migration.
+second message
+last message
+status report
+ OK   [3/3] Migrated the cluster.
+
+ OK   Migration completed successfully.`)
+	})
+}
+
+func startTest_Failure(t *testing.T) {
+	tcx := it.TestContext{T: t}
+	ctx := context.Background()
+	tcx.Tester(func(tcx it.TestContext) {
+		go tcx.WithReset(func() {
+			tcx.CLC().Execute(ctx, "start", "dmt-config", "--yes")
+		})
+		failureRunner(tcx, ctx)
+		tcx.AssertStdoutContains(`
+Hazelcast Data Migration Tool v5.3.0
+(c) 2023 Hazelcast, Inc.
+	
+Selected data structures in the source cluster will be migrated to the target cluster.	
+
+
+ OK   [1/3] Connected to the migration cluster.
+first message
+ OK   [2/3] Started the migration.
+second message
+fail status report
+ FAIL Could not migrate the cluster: migration failed with following error(s):
+error1
+error2`)
+	})
+}
+
+func successfulRunner(tcx it.TestContext, ctx context.Context) {
+	c := make(chan string, 1)
+	go findMigrationID(ctx, tcx, c)
+	migrationID := <-c
+	topic := MustValue(tcx.Client.GetTopic(ctx, migration.MakeUpdateTopicName(migrationID)))
+	Must(topic.Publish(ctx, migration.UpdateMessage{Status: migration.StatusInProgress, Message: "first message"}))
+	Must(topic.Publish(ctx, migration.UpdateMessage{Status: migration.StatusInProgress, Message: "second message"}))
```

Comment:
UpdateMessage should have comp. percentage field always.

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/367/commits/d0dfafa7c167f535e3ba6703862c0d9ea14dc7b6